### PR TITLE
[FEATURE] DataProfilerColumnDomainBuilder

### PR DESCRIPTION
**Pull request by alexjeilert**
Changes proposed in this pull request:

- Updated `DataProfilerColumnDomainBuilder` class to remove profile_path instance variable.
- Updated `DataProfilerColumnDomainBuilder`'s `_get_domains()` function to take in profile_path, metric, and metric_values as a part of variables
- Updated DataProfilerTableColumnList to filter columns from the profile object by checking if the `metric` of the column is in `metric_values`
- Updated `test_data_profiler_column_domain_builder` and `test_data_profiler_structured_data_assistant` to match updated `DataProfilerColumnDomainBuilder` functionality

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.

### CODE REVIEW

File: contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/metrics/data_profiler_metrics/data_profiler_table_column_list.py [hunk 1]
```diff
@@ -37,7 +39,14 @@ def _pandas(
             column_names=profile_report_column_names,
             batch_columns_list=metrics["table.columns"],
         )
-        return profile_report_column_names
+        profile_report_filtered_column_names: list = []
```
Comment (alexsherstinsky) [inline contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/metrics/data_profiler_metrics/data_profiler_table_column_list.py:42]: @alexjeilert If we are filtering in this different way, then do we still need the use of GX metrics for this purpose (the `get_dbms_compatible_column_names()`) call?  Also, the filtering code can be shortened using a list comprehension.  Thank you.

File: contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/rule_based_profiler/domain_builder/data_profiler_column_domain_builder.py [hunk 2]
```diff
@@ -110,18 +103,37 @@ def _get_domains(
         # Obtain profile_path from "rule state" (i.e., variables and parameters); from instance variable otherwise.
         profile_path: str = get_parameter_value_and_validate_return_type(
             domain=None,
-            parameter_reference=self.profile_path,
```
Comment (alexsherstinsky) [inline contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/rule_based_profiler/domain_builder/data_profiler_column_domain_builder.py:113]: @alexjeilert Sounds like you are always going to use the `variables` key `profile_path`, so I can see how your simplification makes sense.  Thanks.

File: contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/rule_based_profiler/domain_builder/data_profiler_column_domain_builder.py [hunk 3]
```diff
@@ -110,18 +103,37 @@ def _get_domains(
...
 
+        metric: str = get_parameter_value_and_validate_return_type(
+            domain=None,
+            parameter_reference=f"{VARIABLES_KEY}metric",
+            expected_return_type=str,
+            variables=variables,
+            parameters=None,
+        )
+
+        metric_values: str = get_parameter_value_and_validate_return_type(
```
Comment (alexsherstinsky) [inline contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/rule_based_profiler/domain_builder/data_profiler_column_domain_builder.py:120]: @alexjeilert The name `metric_values` is not ideal here; I believe these are the parameters of the metric you are going to use, so I would suggest a more fitting name (whereas "metric value" usually refers to the value of a metric that was computed -- this appears to be different).  Thank you.

File: contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/tests/rule_based_profiler/data_assistant/test_data_profiler_structured_data_assistant.py [hunk 4]
```diff
@@ -64,8 +58,9 @@ def bobby_profile_data_profiler_structured_data_assistant_result_usage_stats_ena
                 "data_profiler_files",
                 "profile.pkl",
             ),
+            "metric": "data_type",
+            "metric_values": ["int", "float"],
```
Comment (alexsherstinsky) [inline contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/tests/rule_based_profiler/data_assistant/test_data_profiler_structured_data_assistant.py:62]: @alexjeilert Please see my note above about possibly changing the name `metric_values` to something more intuitive for this purpose.  Thank you.

File: contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/tests/rule_based_profiler/domain_builder/test_data_profiler_column_domain_builder.py [hunk 5]
```diff
@@ -19,7 +19,6 @@
 from great_expectations.data_context import FileDataContext
 from great_expectations.rule_based_profiler.domain_builder import DomainBuilder
 from great_expectations.rule_based_profiler.parameter_container import (
-    VARIABLES_KEY,
```
Comment (alexsherstinsky) [inline contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/tests/rule_based_profiler/domain_builder/test_data_profiler_column_domain_builder.py:22]: @alexjeilert Now with the simplification of the `DomainBuilder`, are any of these tests duplicates, which could be deleted?  Thanks.

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
Please see comments inline.  Once all the lint checks are passing, I would be happy to review again.  Thank you!

**Comment by micdavis** [comment]
The issues regarding this PR have been addressed in #7920. Please close this PR as the changes are being addressed in the other PR. @alexsherstinsky

**Comment by alexsherstinsky** [comment]
Closing per request by @micdavis -- thank you.

**Pull request closed without merging**